src/js/filterRating.js

```diff
@@ -0,0 +1,58 @@
+
+export function handleStarInput(data) {
+
+  const starsFrom = document.querySelectorAll(".starsFrom li");
+  const starsTo = document.querySelectorAll(".starsTo li");
+  
+  let activeIdx = null;
+
+  starsFrom.forEach((starFrom, clickIdx) => {
+    starFrom.addEventListener("click", (e) => {
+      if (activeIdx === clickIdx) {
+        e.target.classList.remove("active");
+        activeIdx = clickIdx - 1;
+      } else {
+        e.target.classList.add("active");
+        activeIdx = clickIdx;
+      }
+  
+      starsFrom.forEach((otherStar, otherIdx) => {
+        if (otherIdx < clickIdx) {
+          otherStar.classList.add("active");
+        } else if (otherIdx > clickIdx) {
+          otherStar.classList.remove("active");
+        }
+      });
+    });
+  });
```

Comment:
I think this code is pretty nice. It uses the `activeIdx` variable as the data store for the new value, and updates the UI to reflect `activeIdx` (although it uses the `clickIdx` variable, which is confusing but still fine).

The next chunk of code is almost identical, but for the maximum rating widget, so maybe there could have been some way to generalize this and reuse code?